qcengine/programs/nwchem/runner.py

```diff
@@ -147,8 +148,31 @@ def build_input(
         # Handle conversion from schema (flat key/value) keywords into local format
         optcmd = format_keywords(opts)
 
+        # Combine the molecule description, options and method command together
         nwchemrec["infiles"]["nwchem.nw"] = "echo\n" + molcmd + optcmd + mdccmd
 
+        # For gradient methods, add a Python command to save the gradients in full precision
+        if input_model.driver == "gradient":
+            # Get the name of the theory used for computing the gradients
+            theory = re.search("^task (\w+) ", mdccmd, re.MULTILINE).group(1)
+            logger.debug(f"Adding a Python task to retrieve gradients. Theory: {theory}")
+
+            # Create a Python function to get the gradient from NWChem's checkpoint file (rtdb)
+            #  and save it to a JSON file. The stdout for NWChem only prints 6 _decimal places_
+            #  (not 6 significant figures)
+            pycmd = f"""
+python
+  if ga_nodeid() == 0:
+      import json
+      grad = rtdb_get('{theory}:gradient')
+      with open('nwchem.grad', 'w') as fp:
+          json.dump(grad, fp)
+end
+
+task python
+            """
+            nwchemrec["infiles"]["nwchem.nw"] += pycmd
```

Comment:
Could we do something similar with the Hessian?

Reply:
We don't need to. The Hessian is already printed to a file in higher precision.

Reply:
Oh neat, can we add a comment that this isn't needed for the Hessian?